backend/api/routes/files.py

```diff
@@ -147,6 +151,52 @@ def validate_project_quota(file_size: int, project: Project):
         )
 
 
+def s3_file_key(project_id: UUID, path: str) -> str:
+    """Generate s3 file key."""
+    return f"{project_id}-{path}"
+
+
+def update_file_status_and_path(file: File, status: FileStatus, path: str):
+    """Update file's Status and Path."""
+    session = next(gen_session())
+    stmt = update(File).filter_by(id=file.id).values(status=status, path=path)
+    session.execute(stmt)
+    session.commit()
+
+
+def upload_file_to_s3(new_file: File):
+    """Update local file to S3 storage and update file status"""
+    s3_key = s3_file_key(new_file.project_id, new_file.hash)
+
+    if s3_storage.storage.has_object(s3_key):
```

Comment:
Do we need to change the status of the file to something like `S3_UPLOADING` before the file is uploaded?

Reply:
> And you should check it first and return should it being uploaded

What does this mean? Can you explain it?

Reply:
Before you actually start upload, you check in DB if status is `UPLOADING` and return if that's the case. If not (status=LOCAL) then you change to UPLOADING and proceed